### ISSUE TITLE
`rake db:setup` fails for new development environment

### ISSUE THREAD

**Issue by icarito** [publiclab/plots2#1392]
**NOTE** some issues may occur on Rails 2.4; please use <=2.3 (see comments below)

### What happened just before the problem occurred
Tried to create development environment.

### Relevant URLs
README.md, install_cloud9.sh

In cloud9 vm, Ubuntu 14.04, impossible to `rake db:setup`
Same thing reported on OS X EL Capitan version 10.11.6, ruby 2.3.1.
Fails with missing mysql2 gem.
Following instructions from repository.
Same issue reported by NikhilAgrawal on chat, log attached.

[log.txt](https://github.com/publiclab/plots2/files/972397/log.txt)

**Comment by icarito** [comment]
```
$ rake db:setup
development.sqlite already exists
test.sqlite already exists
rake aborted!
Gem::LoadError: mysql2 is not part of the bundle. Add it to Gemfile.
```

**Comment by icarito** [comment]
NikhilAgrawal reports same issue on "OS X EL Capitan version 10.11.6"

**Comment by jywarren** [comment]
@nagrawal3 - can you upload your Gemfile.lock also to https://gist.github.com and share it with us so we can debug?

**Comment by nagrawal3** [comment]
@jywarren link to Gemfile.lock https://gist.github.com/nagrawal3/e2a21af5f62a5039de9bd01c1d5031be

**Comment by jywarren** [comment]
Hmm, that's so odd, since `mysql2` is shown in the `Gemfile.lock`. Could you try `bundle exec rake db:setup`? Sometimes you have to get access the gems via bundler explicitly.

**Comment by jywarren** [comment]
And also the output of `gem list` would be helpful!

**Comment by nagrawal3** [comment]
@jywarren I had tried running `bundle exec rake db:setup` but still getting same error.

List of all local gems installed https://gist.github.com/nagrawal3/779bb4c7d8fde152afe49224873e45bd

**Comment by jywarren** [comment]
Hmm, that's odd -- and even this shows `mysql2 (0.4.6)` -- hmmmmm. Let me
look a bit deeper.

**Comment by jywarren** [comment]
And can you tell us:

1. what your database.yml looks like (be sure to remove your passwords!)
2. are you sure you ran `bundle install --without production mysql` as listed in https://github.com/publiclab/plots2/#installation ?

Thank you so much for working through this with us; we really appreciate your patience.

**Comment by jywarren** [comment]
And for reference, the closest I've gotten on a similar issue is: http://stackoverflow.com/questions/9714965/how-to-resolve-please-install-the-mysql-adapter-gem-install-activerecord-mysql#10326485

**Comment by jywarren** [comment]
Just reminder to self that we're working on this in https://ide.c9.io/icarito/plots2-vanilla

**Comment by icarito** [comment]
That c9 instance ran out of space and I damaged the install by removing `~/.local/share/heroku`.
I will start afresh.

**Comment by nagrawal3** [comment]
@jywarren I made sure that I am running the `bundle install --without production mysql` command. I am getting the same error, after doing bundle install in above manner.

I also tried adding `config.assets.initialize_on_precompile = false` in application.rb file as mentioned in the reference you mentioned above (http://stackoverflow.com/questions/9714965/how-to-resolve-please-install-the-mysql-adapter-gem-install-activerecord-mysql#10326485), still getting same error.

Below is the content of my database.yml file:
development:
  adapter: sqlite3
  database: development.sqlite

production:
  adapter: sqlite3
  database: production.sqlite

test:
  adapter: sqlite3
  database: test.sqlite

I created my database.yml file by following the step 5 of the installation instructions. Just letting you know I didn't have any password field in database.yml file.

**Comment by icarito** [comment]
I reproduced this on Debian 8.

**Comment by icarito** [comment]
Okay I managed to run `rake db:setup`!
The issue is in the file `config/initializers/abstract_mysql2_adapter.rb`
This file is requiring MySQL but rightfully not finding it.
It worked for me simply moving the file away and running `rake db:setup`.

**Comment by icarito** [comment]
Just a note that I had previously I had removed and regenerated Gemfile.lock.

**Comment by jywarren** [comment]
Hi all, can you try to pull from master again and try once more? I believe @icarito has solved this!

**Comment by StlMaris123** [comment]
The tests are now running but there is a funny message that is being displayed before a task is run 
 `Not running mysql today.
`

**Comment by nagrawal3** [comment]
Even I am able to see message `Not running mysql today.`

**Comment by icarito** [comment]
Yes, the message is accurate and harmless but if it is distracting we should remove it.

**Comment by biswesh456** [comment]
Hi. I am facing the same problem now when I tried to follow the instructions given in readme.
It asks for bundle exec rake db:setup. On typing the command I see the following error:

/Users/bisweshmohapatra/.rvm/gems/ruby-2.4.1/gems/activesupport-4.1.16/lib/active_support/core_ext/numeric/conversions.rb:121: warning: constant ::Fixnum is deprecated
/Users/bisweshmohapatra/.rvm/gems/ruby-2.4.1/gems/activesupport-4.1.16/lib/active_support/core_ext/numeric/conversions.rb:121: warning: constant ::Bignum is deprecated
rake aborted!
SystemStackError: stack level too deep

**Comment by jywarren** [comment]
Ah, that's not good! 

@StlMaris123 - are you seeing this same issue? 

Can anyone post a full log? I think the warnings may not be related, in fact -- just a suspicion!

**Comment by biswesh456** [comment]
On running the rake db:setup I get this:

rake aborted!
Gem::LoadError: You have already activated rake 12.3.0, but your Gemfile requires rake 10.5.0. Prepending `bundle exec` to your command may solve this.

So on using bundle exec I got the above errors.

On searching a bit I have found that rails version might be creating a problem. Is that a possible problem?

**Comment by biswesh456** [comment]
It worked after I changed my ruby from 2.4.1 to 2.2.0. Thanx for cooperating.

**Comment by jywarren** [comment]
Great, thank you for fixing this. I left a note at the top for reference and will close. We should try to refactor to support Rails 2.4. @StlMaris123 please note the Rails 2.4 incompatibility!

**Comment by jywarren** [comment]
Left a note in #2075

**Comment by alator21** [comment]
I'm having this issue right now.I had the exact same situation as @icarito but after i removed the specific file he mentioned it changed.Now im getting this..


--add_index("comments",["comment"],{:name=>"index_comments_on_comment",:type=>:fulltext})
rake aborted!
ActiveRecord::StatementInvalid: SQLite3::SQLException: near "fulltext": syntax error:CREATE fulltext INDEX "index_comments_on_comment" ON "comments" ("comment")


ruby -v commands gives result ruby 2.3.4p301 ....
any clues?

**Comment by Gauravano** [comment]
hi @alator21 , this problem is addressed by @jywarren in #1997 but now the schema.rb.example is again changed so you can try the changes done by @jywarren here https://github.com/publiclab/plots2/pull/1997/files .I hope, that would solve your problem.

Also, @Ronaq13, you can also try modifying schema.rb.example , if I am not wrong you are facing same problem as faced by @alator21 .

**Comment by jywarren** [comment]
OK, i've merged in @alator21 's great work in #2505, so this should be OK now!
